framework/src/bcs/IntegratedBC.C

```diff
@@ -129,28 +129,22 @@ IntegratedBC::computeJacobian()
 void
 IntegratedBC::computeJacobianBlock(MooseVariableFEBase & jvar)
 {
-  computeJacobianBlock(jvar.number());
-}
-
-void
-IntegratedBC::computeJacobianBlock(unsigned int jvar)
-{
-  if (jvar == _var.number())
+  if (jvar.number() == _var.number())
   {
     computeJacobian();
     return;
   }
 
-  prepareMatrixTag(_assembly, _var.number(), jvar);
+  prepareMatrixTag(_assembly, _var.number(), jvar.number());
 
   // This (undisplaced) jvar could potentially yield the wrong phi size if this object is acting
   // on the displaced mesh
-  auto phi_size = _sys.getVariable(_tid, jvar).dofIndices().size();
+  auto phi_size = jvar.dofIndices().size();
```

Comment:
Did you read this comment before you made this change?

Reply:
ah, no. lol.

Reply:
It is telling you the exact problem that happened

Reply:
`_sys` is the displaced nonlinear system if `use_displaced_mesh` for that object is `true`, else it is the undisplaced nonlinear system

Reply:
There two other places to fix: `NodeFaceConstraint.C ` and `GapHeatTransfer.C`.